subsys/net/lib/mqtt_sn/mqtt_sn.c

```diff
@@ -55,9 +56,23 @@ struct mqtt_sn_topic {
 	enum mqtt_sn_topic_state state;
 };
 
-K_MEM_SLAB_DEFINE_STATIC(publishes, sizeof(struct mqtt_sn_publish),
-			 CONFIG_MQTT_SN_LIB_MAX_PUBLISH, 4);
+struct mqtt_sn_addr {
+	char addr[CONFIG_MQTT_SN_LIB_MAX_ADDR_SIZE];
+	size_t addr_len;
```

Comment:
I was expecting the addresses to be agnostic to the library and it would only store the address and pass pointers back to the transport code the user creates. Does the byte order matter then?

Reply:
Sure, but the address format for a specific transport should still be known for the application, I'm not sure if it's documented anywhere right now.

Reply:
I could add some documentation to that that the user will need to define this based on their implemented transport?

Reply:
That would be nice.